app/src/org/commcare/CommCareApplication.java

```diff
@@ -1206,4 +1209,14 @@ public boolean isNsdServicesEnabled() {
     public void customiseOkHttp() {
         CommCareNetworkServiceGenerator.customizeRetrofitSetup(new OkHttpBuilderCustomConfig());
     }
+
+    private void setRxJavaGlobalHandler() {
+        RxJavaPlugins.setErrorHandler(throwable -> {
+            // Swallow UndeliverableException
+            if (throwable instanceof UndeliverableException) {
+                return;
+            }
+            Thread.currentThread().getUncaughtExceptionHandler().uncaughtException(Thread.currentThread(), throwable);

```

Comment:
Does this deliver the exception to `CommCareExceptionHandler` ?

Reply:
Yes, it does.